src/smtp/SMTPServer.ts

```diff
@@ -39,6 +40,15 @@ export default class SMTPServer {
 			content: ""
 		}
 
+		// State:
+		// 0: No authentication
+		// 1: Waiting for authentication (AUTH PLAIN)
+		// 2: Authenticated
+		const auth = {
+			state: 0,
```

Comment:
can you make this an Enum or at least a Union type, e.g. `"off" | "waiting" | "authenticated"`